core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp

```diff
@@ -290,7 +290,8 @@ KOKKOS_INLINE_FUNCTION
 // Go in this branch if CUDA version is >= 11.0.0 and less than 11.1.0 or if the
 // architecture is not Ampere
 #if CUDA_VERSION >= 11000 && \
-    (CUDA_VERSION < 11010 || !defined(KOKKOS_ARCH_AMPERE))
+    (CUDA_VERSION < 11010 || \
+     !(defined(KOKKOS_ARCH_AMPERE) || defined(KOKKOS_ARCH_HOPPER)))
```

Comment:
Comment above mis-matches.

As a potential refactoring, should we use some sort of define for CUDA compute capability, so we can durably say (e.g.) `KOKKOS_ARCH_CUDA_COMPUTE < 80` rather than having to update the condition for each *new* architecture for which we add support?